replicaset.sh

```diff
@@ -70,43 +79,47 @@ fi
 
 log "Peers: ${peers[*]}"
 
-log "Starting a MongoDB instance..."
-mongod --config /data/configdb/mongod.conf --dbpath=/data/db --replSet="$replica_set" --port=27017 "${auth_args[@]}" --bind_ip=0.0.0.0 2>&1 | tee -a /work-dir/log.txt &
-
 log "Waiting for MongoDB to be ready..."
 until mongo --host localhost "${ssl_args[@]}" --eval "db.adminCommand('ping')"; do
-    log "Retrying..."
+    log "Retrying to ping..."
     sleep 2
 done
 
 log "Initialized."
 
-# try to find a master and add yourself to its replica set.
-for peer in "${peers[@]}"; do
-    if mongo admin --host "$peer" "${admin_creds[@]}" "${ssl_args[@]}" --eval "rs.isMaster()" | grep '"ismaster" : true'; then
-        log "Found master: $peer"
-        log "Adding myself ($service_name) to replica set..."
-        mongo admin --host "$peer" "${admin_creds[@]}" "${ssl_args[@]}" --eval "rs.add('$service_name')"
+# check if the replica is already added in replicaset
+if [[ $(mongo admin --host localhost "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "rs.status().myState") == '2' ]]; then
+    log "($service_name) already added in replicaset"
+    log "Good bye."
+    exit 0
+else
+    # try to find a master and add yourself to its replica set.
+    for peer in "${peers[@]}"; do
+        if mongo admin --host "$peer" "${admin_creds[@]}" "${ssl_args[@]}" --eval "rs.isMaster()" | grep '"ismaster" : true'; then
+            log "Found master: $peer"
 
-        sleep 3
+            log "Adding myself ($service_name) to replica set..."
+            retry mongo admin --host "$peer" "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "JSON.stringify(rs.add('$service_name'))"
 
-        log 'Waiting for replica to reach SECONDARY state...'
-        until printf '.' && [[ $(mongo admin --host localhost "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "rs.status().myState") == '2' ]]; do
-            sleep 1
-        done
+            sleep 3
```

Comment:
All these `sleep 3`. Should we put a `DEFAULT_WAIT_SECS=3` at the top, then `sleep $DEFAULT_WAIT_SECS`